fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/KeyExpression.java

```diff
@@ -230,6 +231,38 @@ public static FanType valueOf(RecordMetaDataProto.Field.FanType fanType) throws
     @Nonnull
     KeyExpression normalizeForPlanner(@Nonnull Source source, @Nonnull List<String> fieldNamePrefix);
 
+    /**
+     * Return the key fields for an expression.
+     * @param rootExpression a key expression
+     * @return the parts of the key expression that are in the key
+     */
+    @API(API.Status.EXPERIMENTAL)
+    static List<KeyExpression> getKeyFields(@Nonnull KeyExpression rootExpression) {
```

Comment:
John had it that way but I think this is preferable. The awkward thing about the key fields/value fields distinction is that it's really only conceptually defined on the root expression (which might be a `KeyWithValueExpression`).

I have an old issue somewhere about splitting the `KeyExpression` interface into a `KeyExpression` (which is a true tree-like data structure) and `IndexExpression` (which might be a `KeyWithValueExpression`, `GroupingExpression`, or  just a plain `KeyExpression`), but obviously that's a highly invasive change.

Given that, I'd rather keep these as static methods, since that seems to contain the "conceptual bleed". I'm willing to change it up if there's strong disagreement, though.

Reply:
Fine either way.